server/sandbox_network.go

```diff
@@ -21,7 +21,7 @@ import (
 func (s *Server) networkStart(ctx context.Context, sb *sandbox.Sandbox) (podIPs []string, result cnitypes.Result, retErr error) {
 	overallStart := time.Now()
 	// give a network Start call 2 minutes, half of a RunPodSandbox request timeout limit
-	startCtx, startCancel := context.WithTimeout(ctx, 2*time.Minute)
+	startCtx, startCancel := context.WithTimeout(ctx, 4*time.Minute)
```

Comment:
Why? :-)

Mind to add an explanation in the commit message and also update the comment above (which still mentiones 2 minutes).

Reply:
Looking at:
```
resourcecache: add watcher idiom

Even if we use the resource cache as is, the user is still bombarded with messages saying the name is reserved.

This is bad UX, and we're capable of improving it.

Add watcher idiom to resource cache, allowing a handler routine of RunPodSandbox or CreateContainer to
wait for a resource to be available.

Something that is key here is if the resource becomes available while we're watching for it,
*we still need to error on this request*
This is because we could get the resource from the cache, remove it (thus meaning it won't be cleaned up),
and the kubelet's request could time out, and it could try again. This would cause us to leak a resource.

This way, if we get into this situation, there needs to be three requests:
first that times out
second that discovers the resource is ready, but still errors
third that actually retrives that resource and returns it.

This will result in many fewer "name is reserved" errors (one every 2 seconds to one every 4 minutes)
```

Is this the reason of updating it?

Reply:
updated comment and commit, thanks!